pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/stats/LongColumnPreIndexStatsCollector.java

```diff
@@ -100,8 +97,11 @@ public int getCardinality() {
 
   @Override
   public void seal() {
-    _sortedValues = _values.toLongArray();
-    Arrays.sort(_sortedValues);
-    _sealed = true;
+    if (!_sealed) {
```

Comment:
If there is no sorting after seal, we should also throw exception in collect(Object) if _sealed=true to avoid any unexpected behavior.

Reply:
I don't want to introduce an extra per-value check (other checks are per column). IMO it is okay for it to throw NPE if someone uses it incorrectly